src/SplitClient.tsx

```diff
@@ -114,8 +115,9 @@ export class SplitComponent extends React.Component<IUpdateProps & { factory: Sp
   }
 
   render() {
-    const { children } = this.props;
-
+    const { children, client, attributes } = this.props;
+    if (attributes)
```

Comment:
2 things to consider about the call to setAttributes:
- It is better if we do it inside `SplitComponent`, to reuse the logic for SplitFactory component. Probably inside `SplitComponent::getDerivedStateFromProps` method is okey, since `setAttributes` is an idempotent operation (i.e., executing it multiple times is equivalent to executing it once).
- we must also consider that `attributes` might have some values at the beginning, and then be `undefined` on a component update. So we might probably need to call `client.setAttributes` or `client.deleteAttributes` if attributes is undefined.

Reply:
`SplitComponent::getDerivedStateFromProps` is the correct place for that logic. `SplitComponent::render` is also fine, but it is better the other way for separation of concerns.